cmd/astro-wasm/astro-wasm.go

```diff
@@ -202,15 +202,22 @@ func ConvertToTSX() interface{} {
 		transformOptions := makeTransformOptions(js.Value(args[1]), "XXXXXX")
 
 		var doc *astro.Node
-		doc, err := astro.Parse(strings.NewReader(source))
+		doc, err := astro.Parse(strings.NewReader(strings.TrimRightFunc(source, unicode.IsSpace)))
```

Comment:
what is this doing (just curious, not questioning)

Reply:
if this is trimming spaces could that mess with locations?

Reply:
This just trims any trailing newlines from the source code before we process it. This avoid an extra `TextNode` at the end of the file. Doesn't break any mapping AFAICT.